addon/-private/normalize-object.ts

```diff
@@ -0,0 +1,29 @@
+import isObject from '../utils/is-object';
+
+/**
+ * traverse through target and unset `value` from leaf key
+ * Shallow copy here is fine because we are swapping out the leaf nested object
+ * rather than mutating a property in something with reference
+ *
+ * @method normalizeObject
+ * @param target
+ */
+export default function normalizeObject<T extends { [key: string]: any}>(target: T): T {
+  let obj = { ...target };
+
+  for (let key in obj) {
+    if (key === 'value') {
```

Comment:
Can we get this check to be the first thing to do?

```js
if ('value' in target) { // Or hasOwnProperty
  return target.value;
}
```

That way we prevent creating a new object and traversing any property.